src/main/java/com/khutircraftubackend/security/JwtUtils.java

```diff
@@ -0,0 +1,59 @@
+package com.khutircraftubackend.security;
+
+import io.jsonwebtoken.*;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+
+import java.time.Instant;
+import java.util.Date;
+
+/**
+ * Клас JwtUtils містить методи для створення та перевірки JWT токенів.
+ */
+
+@RequiredArgsConstructor
+@Slf4j
+public class JwtUtils {
+
+    @Value("${jwt.secret}")
+    private String jwtSecret;
+
+    @Value("${jwt.expiration}")
+    private int jwtExpirationSec;
+
+    public String generateJwtToken(String email) {
+        String jwt = Jwts.builder()
+                .setSubject(email)
+                .setIssuedAt(Date.from(Instant.now()))
+                .setExpiration(Date.from(Instant.now().plusSeconds(jwtExpirationSec)))
+                .signWith(SignatureAlgorithm.HS256, jwtSecret)
+                .compact();
+        log.info("Generate JWT: {}", jwt);
+        return jwt;
+    }
+
+    public String getJwtUsernameFromToken(String token) {
+        return Jwts.parser()
+                .setSigningKey(jwtSecret)
+                .parseClaimsJws(token)
+                .getBody()
+                .getSubject();
+    }
+
+    public boolean validateJwtToken(String token) {
+        try {
+            Jwts.parser().setSigningKey(jwtSecret).parseClaimsJws(token);
+            return true;
+        } catch (SignatureException e) {
+            log.error("Invalid JWT signature: {}", e.getMessage());
```

Comment:
should REST API client know abt the exact error?

Reply:
add exception